alembic/versions/3a677cb88b01_create_algorithm_tables.py

```diff
@@ -0,0 +1,53 @@
+"""create algorithm tables
+
+Revision ID: 3a677cb88b01
+Revises: 6052c193a26a
+Create Date: 2024-09-19 17:07:41.379176
+
+"""
+from typing import Sequence, Union
+
+from alembic import op
+import sqlalchemy as sa
+
+
+# revision identifiers, used by Alembic.
+revision: str = '3a677cb88b01'
+down_revision: Union[str, None] = '6052c193a26a'
+branch_labels: Union[str, Sequence[str], None] = None
+depends_on: Union[str, Sequence[str], None] = None
+
+
+def upgrade() -> None:
+    # ### commands auto generated by Alembic - please adjust! ###
+    op.create_table('algorithm',
+    sa.Column('id', sa.Integer(), nullable=False),
+    sa.Column('is_default', sa.Boolean(), nullable=False),
+    sa.Column('label', sa.String(length=255), nullable=False),
+    sa.Column('description', sa.Text(), nullable=False),
+    sa.PrimaryKeyConstraint('id'),
+    sa.UniqueConstraint('label')
+    )
+    op.create_index(op.f('ix_algorithm_is_default'), 'algorithm', ['is_default'], unique=False)
+    op.create_table('algorithm_pass',
+    sa.Column('id', sa.Integer(), nullable=False),
+    sa.Column('algorithm_id', sa.Integer(), nullable=False),
+    sa.Column('blocking_keys', sa.JSON(), nullable=False),
+    sa.Column('evaluators', sa.JSON(), nullable=False),
```

Comment:
#13 indicates that the `blocking_keys` column should be list of int and that the `evaluators` column should be list of str. What was the reason for the change to JSON type?

Reply:
Arrays/Lists aren't a standard Type in SQL.  This is really by design, since traditional database schemas require new tables for one->many relationships.  So 20 years ago we would have needed to create a `algorithm_pass_blockingkey` and `algorithm_pass_evaluator` child tables to store this.


We could always go that route, but for small datasets, I find it less readable (you need to go back and forth between tables, you always have to join tables when querying, you have to remember to add order columns to preserve order).  Lists kind of make this easier.

While JSON is becoming more commonplace, postgresql, mysql, microsoft sql server, sqlite3 all support this.  ARRAY is not.  So for maximum database support, using JSON gives us more flexibility.

Reply:
Postgresql does support Array datatypes but I made it JSON for the same reasons Eric said above^